### ISSUE TITLE
Results broken/incorrect when search yields no match

### ISSUE THREAD

**Issue by bengm** [18F/calc#553]
The search results view seems to essentially break when no matching records are found after searching for a term with no hits.

## Steps to Replicate

1) Search for something with hits (e.g. "programmer"), get results:
![image](https://cloud.githubusercontent.com/assets/819222/17649448/986e0128-6203-11e6-90c0-9ef314a4f18e.png)

2) Search again for something with no match: type in "agile coach"; hit Search button:
![image](https://cloud.githubusercontent.com/assets/819222/17649454/cd6a8a36-6203-11e6-8d51-1988ba184489.png)

The results are now wonky. For example, "Showing 0 of 706 results with agile coach" and then a bar chart and below detail table with data based on old search.

Note: may be related to #295

**Comment by toolness** [comment]
Thanks for the extensive bug report, @bengm !  Are you using calc.gsa.gov?  If so, this may actually be a duplicate of #244, which was fixed in #294--however, we have only been able to deploy the most critical security fixes to calc.gsa.gov because we're waiting on an [ATO](https://pages.18f.gov/before-you-ship/ato/) for new features that are being added to CALC in its next phase. Unfortunately, #294 wasn't deemed a "critical fix" so it hasn't been deployed to calc.gsa.gov yet. 😞 

Can you try the same thing on https://calc-dev.apps.cloud.gov/ ?  Does that one work for you?

**Comment by toolness** [comment]
I'm going to close this bug now because I'm pretty sure it's fixed and we'll push it to production soon--feel free to reopen it if you disagree, though!  Thanks again for the report.

**Comment by bengm** [comment]
Yes, seems to work fine for me on the dev site.  Sorry for the delay, notification was buried :)
